code/modules/power/apc/apc_main.dm

```diff
@@ -451,6 +451,9 @@
 		main_status = APC_NO_POWER
 	else if(excess < 0)
 		main_status = APC_LOW_POWER
+		for(var/obj/machinery/light/flickering_light in area)
```

Comment:
You should never loop `for(thing in area)`, what it does internally is loop over every single thing in the *entire world*, and checks if its in the area. I believe you can just change this to `area.get_contained_turfs()`

Reply:
We'd lint for this if our tooling was better

Reply:
More specifically, you loop over every turf, then loop over every light, but this is still going to be pretty slow...this should probably be a signal on the area instead that lights listen for. That way we can even avoid the CHECK_TICK that I am worried about

Reply:
cant we just grep for "in area)" then call it a day

Reply:
You would get some but not nearly all of them, my actual plan for this was finding the code path in BYOND and setting a detour there to log when it gets hit lol